src/main/java/org/openstreetmap/atlas/geography/atlas/pbf/loading/OsmPbfReader.java

```diff
@@ -0,0 +1,420 @@
+    }
+
+    /**
+     * First, creates an {@link Entity} {@link Tag} for specific OSM attributes we're interested in
+     * propagating to the {@link AtlasEntity}. Secondly, converts the given {@link Entity}'s
+     * collection of {@link Tag}s to a {@link Map} of key/value pairs used to build an
+     * {@link AtlasEntity}'s tag set.
+     *
+     * @param entity
+     *            The {@link Entity} being processed
+     * @return a {@link Map} of key/value tags
+     */
+    private Map<String, String> populateEntityTags(final Entity entity)
+    {
+        // Update the entity's tags to contain specific OSM attributes we care about, so that these
+        // will get translated to Atlas Entity tags.
+        storeOsmEntityAttributesAsTags(entity);
+
+        return new TagMap(entity.getTags()).getTags();
+    }
+
+    /**
+     * Uses the {@link Node} OSM identifier, geometry and tags to create an Atlas {@link Point}.
+     *
+     * @param entity
+     *            The {@link Entity} that will become an Atlas {@link Point}
+     */
+    private void processNode(final Entity entity)
+    {
+        final Node node = (Node) entity;
+        this.builder.addPoint(node.getId(), new Location(Latitude.degrees(node.getLatitude()),
+                Longitude.degrees(node.getLongitude())), populateEntityTags(node));
+    }
+
+    /**
+     * Tries to create an Atlas {@link org.openstreetmap.atlas.geography.atlas.items.Relation}. If
+     * the {@link Entity} contains a member that's also a relation and that member hasn't been
+     * processed yet, then we add the given Relation to a Collection of staged relations to process
+     * later (see {@link #release()} method). Otherwise, we do one last check to make sure the
+     * Relation isn't empty and add it. A relation could be empty if all of its members haven't been
+     * processed yet or are not contained in the given PBF file.
+     *
+     * @param entity
+     *            The {@link Entity} that will become an Atlas
+     *            {@link org.openstreetmap.atlas.geography.atlas.items.Relation}
+     */
+    private void processRelation(final Entity entity)
+    {
+        final Relation relation = (Relation) entity;
+        if (containsUnindexedSubRelation(relation))
+        {
+            // Stage this Relation, it has a member relation that we haven't processed yet
+            this.stagedRelations.add(relation);
+        }
+        else
+        {
+            final RelationBean bean = constructRelationBean(relation);
+            if (!bean.isEmpty())
+            {
+                this.builder.addRelation(relation.getId(), relation.getId(), bean,
+                        populateEntityTags(relation));
+            }
+            else
+            {
+                logger.debug(
+                        "Empty Relation {} cannot be added to the Atlas. We're either filtering out the members that make up the Relation or none of the members are present in this PBF shard.",
```

Comment:
Maybe break that long string into two or three lines?

Reply:
Good call!